contractcourt/channel_arbitrator.go

```diff
@@ -3257,3 +3293,30 @@ func (c *ChannelArbitrator) cancelIncomingHTLCs(
 
 	return nil
 }
+
+// cancelDustHTLCs cancels back the incoming dust htlcs for their corresponding
+// outgoing htlc and also persists the canceled htlcs to the database.
+func (c *ChannelArbitrator) cancelDustHTLCs(dustHTLCs []channeldb.HTLC) error {
+	dustHTLCSet := fn.SliceToMap(dustHTLCs,
+		func(htlc channeldb.HTLC) uint64 {
+			return htlc.HtlcIndex
+		},
+		func(htlc channeldb.HTLC) struct{} {
+			return struct{}{}
+		},
+	)
+
+	err := c.cancelIncomingHTLCs(dustHTLCSet)
```

Comment:
Please check me on this - I think the main reason we want to have the canceled HTLCs persisted in db is to make sure that we don't cancel too much, which is vital for the PR where we decide to cancel non-dust HTLCs. This is why I think we can move the db-related commits there. On the other hand, I think it's fine if we cancel dust HTLCs multiple times as the remote would just ignore them?

Reply:
I think not quite, so I introduced this db entry to avoid loglines like this, which might happen in the htlc_timeout_resolver:

```
[ERR] HSWC: Unable to forward resolution msg: unable to find target channel for HTLC fail: channel ID = 443:2:0, HTLC ID = 0
```

which basically happens if you cancel back htlcs more than once.

I also introduced it in the dustHTLC function to make it universal. I know that we call it in our code, where we would basically never all this code part twice for the same channel, but I did not want to rely on this, and make the cancel function more resilient.